extensions/wikidata/src/org/openrefine/wikidata/editing/EditBatchProcessor.java

```diff
@@ -182,14 +184,31 @@ public void performEdit()
 	                List<MonolingualTextValue> descriptions = update.getDescriptions().stream().collect(Collectors.toList());
 	                descriptions.addAll(update.getDescriptionsIfNew().stream()
 	                        .filter(desc -> !currentDocument.getDescriptions().containsKey(desc.getLanguageCode())).collect(Collectors.toList()));
-	                editor.updateTermsStatements(currentDocument,
-                            labels,
-	                        descriptions,
-	                        update.getAliases().stream().collect(Collectors.toList()),
-	                        new ArrayList<MonolingualTextValue>(),
-	                        update.getAddedStatements().stream().collect(Collectors.toList()),
-	                        update.getDeletedStatements().stream().collect(Collectors.toList()),
-	                        summary, tags);
+	                /* solution 1 */
+	                Set<MonolingualTextValue> aliases = update.getAliases();//set of pairs of strings we're going to add
+	                AliasUpdate aliasUpdate;//what's the next step for 'aliasUpdate'?
+	                Map<String,AliasUpdate> aliasMap = Collections.emptyMap();
+	                for (MonolingualTextValue a: aliases) {
+                        aliasUpdate = aliasMap.get(a.getLanguageCode());
+	                }
+	                /* solution 2
+	                 * Getting same error as with previous attempts:
+	                 * Type mismatch: cannot convert from Map<Object,Object> to Map<String,AliasUpdate>
+	                 *
+	                Set<MonolingualTextValue> aliases = update.getAliases();//set of pairs of strings we're going to add
+	                Map<String, List<MonolingualTextValue>> aliasesMap = aliases.stream().collect(Collectors.groupingBy(MonolingualTextValue::getLanguageCode));
+	                Map<String, List<MonolingualTextValue>> map = new HashMap<>();
+	                Map<String, AliasUpdate> aliasMap = map.entrySet().stream().collect(Collectors.toMap(Entry::getKey, e -> String.valueOf(e.getLanguageCode())));
```

Comment:
Solution 2 is indeed invalid as things stand because your expression `String.valueOf(e.getLanguageCode())` does not produce an `AliasUpdate` but a `String`.

Given the variable `e` of type `List<MonolingualTextValue>`, you need to produce a corresponding `AliasUpdate` object.
This can be done with `Datamodel.makeAliasUpdate(e, Collections.emptyList())`.